R/utils.R

```diff
@@ -0,0 +1,15 @@
+## Until the future package provides a well-defined *apply() API
+## we define futurized versions of lapply() and apply() here.
```

Comment:
There are a few reasons, but first of all, it's really easy to use futures with good-old for loops (that's the original/main design), so no need to pre-maturely specify an API.  Until a good API is settled upon, it's easy enough, like in your case, to define local light-weight utility functions.

To identify/freeze an `*apply()` API, various possible usages of the future package needs to be identified first.  For instance, some people may want to just have a futurized version of `lapply()`, just as with `mcapply()`, `parLapply()` and so on.  Other people might want to be able to pass in existing futures and return new ones ("piping futures").  The latter is something I believe will be really handy and I can only guess some of it's use cases.

I haven't really started a public discussion thread on these topics, but I have a place holder over at https://github.com/HenrikBengtsson/future/issues/21